### PR TITLE
Block screenshots

### LINKED ISSUE #50: Security Problem: Screenshot Function
Original [issue 402](https://code.google.com/p/google-authenticator/issues/detail?id=402) created by TriplexAccount.P.N on 2014-07-08T10:27:41.000Z:

Copied from https://github.com/google/google-authenticator/issues/401 because it applies to Android app (as well?).

<b>What steps will reproduce the problem?</b>
1. Call Screenshot Function from the specific smartphone

<b>What version of the product are you using? On what operating system?</b>
Google Authentificator: 2.49

<b>Please provide any additional information below.</b>
In general, it is not possible to take a screenshot of any inner user interface from a banking app for security reason. The Google Authentificator can be comprimised by using a trojan with screenshot function. Please disable the ability to take a screenshot from the main interface from the Google Authentificator.

### PR DESCRIPTION
**Pull request by Tethik**
## Change Description

This adds a toggleable setting to block screenshots from the main AuthenticatorActivity. By default that setting is set to `true`. Fixes #50 

Unfortunately I haven't been able to run the tests as they make my poor laptop go OOM. I tested it on a test device though. I've also had to upgrade the build tools etc to get the project to run, but I omitted it from this PR to keep it contained.

### CODE REVIEW

**Comment by ThomasHabets** [comment]
Thank you! I'll try to find the best person to review this, and make sure it gets into the Play Store version.

I'm bouncing round the world at the moment, so apologies for delays.

**Comment by Tethik** [comment]
No worries. I'm actually aiming to implement another feature (folders), so I took this one from the issues as a sample to get into the code. From my end there is no rush to release this :)

**Comment by frankenstein91** [comment]
Since according to the German press there is a malicious code in the works that is supposed to take advantage of this missing protective function, but we can take the wind out of the sails with the combined power of OpenSource... what help do @ThomasHabets  need from the community? I think we could find a good helper for each area.

**Comment by ThomasHabets** [comment]
Partially copying this from [the other bug](https://github.com/google/google-authenticator/issues/401)

The issue recently in press recently is, as I understand it, entirely about accessibility functionality that can't (?) be disabled (and for good reason, because accessibility), not about this issue which is about screenshots.

Also, for other people coming here from [ZDNet](https://www.zdnet.com/article/google-could-have-fixed-2fa-code-stealing-flaw-in-authenticator-app-years-ago/):

FYI: The version in Google Play Store / Apple App store is not the same as this opensource version. They've diverged. This opensource version is also unlikely to end up in the app stores. This open source version doesn't get much love, but I'll accept well-written pull requests.

In other words: This bug does NOT track the issue describe in the article, for two reasons:

1. This bug is about screenshots, which AFAIK is not the same issue
2. This repo does not contain the code for Google Authenticator that you can find in any app store what-so-ever

So guess, @frankenstein91, what the community can do is to confirm what API exactly is the relevant one. This pull request seems to disable screenshots, yes, but does it do anything at all to the risk mentioned in the press with this malware? "They" tell me no, it won't.

**Comment by frankenstein91** [comment]
I didn't know about the split, sorry.

I found this article in the online magazine https://www.golem.de/news/google-authenticator-2fa-codes-lassen-sich-einfach-abgreifen-2003-147119.html. Since the second link led into this software, I thought it could be solved by the already opened request. 

I think the article, which at least I read, is only about the screenshot function.

**Comment by Diesmo** [comment]
> Also, for other people coming here from [ZDNet](https://www.zdnet.com/article/google-could-have-fixed-2fa-code-stealing-flaw-in-authenticator-app-years-ago/):
> 
> In other words: This bug does NOT track the issue describe in the article, for two reasons:
> 
> 1. This bug is about screenshots, which AFAIK is not the same issue

I'm a bit confussed, you are talking about the ZDNet article and how the security flaw they describe over there is not relevant to this PR/Screenshot function, but then again this is what the ZDNet article which you linked says:

> If an account was protected by 2FA, and namely by the Google Authenticator app, the malware was designed to allow the Cerberus gang to connect to a user's device manually, via its RAT features. Hackers would then open the Authenticator app, generate one-time passcodes, **take a screenshot of the codes**, and then access the user's account.

For me this sounds exactly like the problem that the PR approaches. 
Removal of screenshot function inside a 2FA app so Malware can't send the codes somewhere.

And reading six digit numbers from a picture is not hard by any means, there are more than enough pre-trained models for this task, which can be set up and running in 10 mins.

File: java/com/google/android/apps/authenticator/AuthenticatorActivity.java [hunk 1]
```diff
@@ -126,6 +127,12 @@
    */
   public static final String KEY_ONBOARDING_COMPLETED = "onboardingCompleted";
 
+  /**
+   * Key under which {@link #onboardingCompleted} is stored to know if user has completed the first
```
Comment (Daniel15) [inline java/com/google/android/apps/authenticator/AuthenticatorActivity.java]: Nit: this comment is incorrect (copy-pasted I guess)

Comment (Tethik) [reply java/com/google/android/apps/authenticator/AuthenticatorActivity.java]: You're totally right, will fix :)
